### ISSUE TITLE
DB/Quest: All Hail Roanauk!

### ISSUE THREAD

**Issue by citizenwow** [TrinityCore/TrinityCore#23917]
Description:

In quest https://www.wowhead.com/quest=12140/ There are no phrases completely and it is impossible to pass the quest. https://www.wowhead.com/npc=26810/

Branch(es): 3.3.5

TC rev. hash/commit:

TrinityCore rev. c79571a 2019-11-03 17:00:24 +0100

Operating system: CentOS 7

**Comment by dr-j** [comment]
Cannot reproduce take quest from overlord agmar he will say line and walk to where roanauk is when agmar is there he will be display gossip if you add quest with command or try to do at a later date this will not work because agmar wont be near him this is a condition of the gossip that agmar is near by

**Comment by citizenwow** [comment]
I take this quest without any command.
![Без названия](https://user-images.githubusercontent.com/57755620/68990103-2f335f80-0860-11ea-8832-239182719090.png)

**Comment by dr-j** [comment]
Overlord agmar has got to be close by for the option to be displayed as said he starts wp on quest accept if you go to roanauk before agmar gets there he won’t display option

**Comment by dr-j** [comment]
https://youtu.be/m9IkxjhAuqc

As said works fine if you try speaking while overlord agmar is too far away he wont display gossip option when you speak to Roanauk Overlord agmar has to be within 30 yards of him otherwise wont display gossip option for event

**Comment by Killyana** [comment]
I remember I case where the quest could be stuck (if the event starts and no one talk to Roanauk ), also some minor visuals must be fixed.

**Comment by Rushor** [comment]
so adding an evade after some minutes should fix that (+active flag)

**Comment by Killyana** [comment]
Something like:
``` sql
DELETE FROM `smart_scripts` WHERE `entryorguid`=26379 AND `source_type`=0 AND `id` IN (5,6,7,8,9,10);
INSERT INTO `smart_scripts` (`entryorguid`, `source_type`, `id`, `link`, `event_type`, `event_phase_mask`, `event_chance`, `event_flags`, `event_param1`, `event_param2`, `event_param3`, `event_param4`, `action_type`, `action_param1`, `action_param2`, `action_param3`, `action_param4`, `action_param5`, `action_param6`, `target_type`, `target_param1`, `target_param2`, `target_param3`, `target_x`, `target_y`, `target_z`, `target_o`, `comment`) VALUES
(26379,0,5,6,38,0,100,0,1,1,0,0,53,0,2637901,0,0,0,2,1,0,0,0,0,0,0,0,"Overlord Agmar - On Data Set - Start wp"),
(26379,0,6,0,61,0,100,0,0,0,0,0,22,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Overlord Agmar - On Data Set  - Set Event phase 0"),
(26379,0,7,8,40,0,100,0,10,2637901,0,0,81,3,0,0,0,0,0,1,0,0,0,0,0,0,0,"Overlord Agmar - On WP10 reached  - set npc flag 3"),
(26379,0,8,0,61,0,100,0,0,0,0,0,24,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Overlord Agmar - On WP10 reached  - evade"),
(26379,0,9,0,40,0,100,0,10,26379,0,0,22,1,0,0,0,0,0,1,0,0,0,0,0,0,0,"Overlord Agmar - On reached WP10 - Set Event phase 1"),
(26379,0,10,0,1,1,100,1,120000,120000,0,0,41,0,0,0,0,0,0,1,0,0,0,0,0,0,0,"Overlord Agmar - Ooc phase 1 - Despawn");


DELETE FROM `waypoints` WHERE `entry` IN (2637901);
INSERT INTO `waypoints` (`entry`, `pointid`, `position_x`, `position_y`, `position_z`, `point_comment`) VALUES
(2637901, 1, 3870.23, 1589.65, 89.759, "Overlord Agmar"),
(2637901, 2, 3868.92, 1577.62, 90.238, "Overlord Agmar"),
(2637901, 3, 3865.92, 1568.37, 90.238, "Overlord Agmar"),
(2637901, 4, 3859.42, 1559.87, 90.238, "Overlord Agmar"),
(2637901, 5, 3850.17, 1547.62, 90.238, "Overlord Agmar"),
(2637901, 6, 3842.67, 1535.87, 90.238, "Overlord Agmar"),
(2637901, 7, 3842.17, 1520.62, 91.738, "Overlord Agmar"),
(2637901, 8, 3842.67, 1511.37, 91.988, "Overlord Agmar"),
(2637901, 9, 3842.42, 1498.37, 91.988, "Overlord Agmar"),
(2637901, 10, 3842.12, 1492.08, 91.717, "Overlord Agmar");

**Comment by Killyana** [comment]
LOL, evades has no effect in npc that are not in combat.

**Comment by dr-j** [comment]
And he can’t evade as soon has he gets to wp since player then needs to select gossip option on Roanoak for which he must be present to show and then event runs which triggers text on agmar so he must not evade on reaching last wp better to use timed despawn and respawn in a short time

**Comment by LKArthas** [comment]
The way i handled was on last wp reached run TAL with a tele to its db posición add its bytes1 and reset npcflag with a time interval of 0 so isn't visible for any player that he do the teleport and @Killyana on the overlord sai add an event 38 to remove event_phase 1 when roanuk scripts ends so you can avoid he runs the back wp unnecesary

**Comment by Killyana** [comment]
Yes I will add a set data on Overlord Agmar  on Roanuk  gossip action, but I want the reset position and bytes1 to be done via evade.

**Comment by LKArthas** [comment]
Got an idea for it how about action 24 add a param1 =1 will be for a walking evade? A posible param2 to evade with pathfinding 0 and 1 to evade withouth pathfinding, i guess the Evades could be run more smoothly like that

**Comment by Killyana** [comment]
It will be great (I thought about it yesterday), but before that we need to fix the evade not triggering if the npc is not in combat

**Comment by dr-j** [comment]
Evade will trigger on a ooc npc that has not followed waypoints ie if you use move to position or move offset but not when using wp entering and leaving combat will make it evade but if moved with other than wp will evade without combat

**Comment by SeekingAura** [comment]
> I remember I case where the quest could be stuck (if the event starts and no one talk to Roanauk ), also some minor visuals must be fixed.

could help if set on npc force Despawn and respawn (after some time) when smart_script with id 4 are executed. This is when Overlord Agmar reach point at Roanauk Icemist.

The next query set force to despawn 120 seconds. Full event when player talk with Roanauk Icemist take less than one minute.
```sql
INSERT INTO smart_scripts (`entryorguid`,`source_type`,`id`,`link`,`event_type`,`event_phase_mask`,`event_chance`,`event_flags`,`event_param1`,`event_param2`,`event_param3`,`event_param4`,`event_param5`,`action_type`,`action_param1`,`action_param2`,`action_param3`,`action_param4`,`action_param5`,`action_param6`,`target_type`,`target_param1`,`target_param2`,`target_param3`,`target_param4`,`target_x`,`target_y`,`target_z`,`target_o`,`comment`) VALUES
(26379,	0,	7,	0,	61,	0,	100,	512,	0,	0,	0,	0,	0,	41,	120000,	3,	0,	0,	0,	0,	1,	0,	0,	0,	0,	0,	0,	0,	0,	"Overlos Agmar - Force Despawn and Respawn On quest All Hail Roanauk! 2 minutes after WP10 is reached");

UPDATE smart_scripts
SET link=7
WHERE entryorguid=26379 AND ID=4;`
```

**Comment by Aokromes** [comment]
anything here is valid?

**Comment by SeekingAura** [comment]
It was a long time ago, and it occurred with an older version of Trinity. It would be best to verify this with a current version.

My suggested solution is to prevent the issue, as I encountered a similar problem in the past.

At that time (when i wrote about this) in my case, the issue occurs when a player interacts with the NPC while they are walking at the event; the NPC will get stuck there without reset or respawn
